fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/IndexKeyValueToPartialRecord.java

```diff
@@ -117,7 +119,6 @@ public void copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull Mess
             if (value == null) {
                 return;
             }
-            final Descriptors.FieldDescriptor fieldDescriptor = recordDescriptor.findFieldByName(field);
```

Comment:
I do not think that a `final` field descriptor will work with currently allowed use cases where the plan is cached and the metadata deserialized from the meta-data store anew. But a one-deep cache can probably get much of the same benefit. Keep the last `recordDescriptor.findFieldByName(field)`; compare its `getContainingType` next time and reuse if match, else replace.